pkg/report/report.go

```diff
@@ -97,11 +104,11 @@ func NewReporter(cfg *mgrconfig.Config) (Reporter, error) {
 	}
 	ctor := ctors[typ]
 	if ctor == nil {
-		return nil, fmt.Errorf("unknown OS: %v", typ)
+		return Reporter{}, fmt.Errorf("unknown OS: %v", typ)
```

Comment:
Since this is a struct now, this need to return a pointer to Reporter now.

Reply:
Is there a particular reason we could prefer pass-by-pointer to pass-by-value w.r.t. this structure?

Reply:
In short because this is Entity type rather than Value type. I am not sure what is a good canonical reference, but here are some:
https://forum.golangbridge.org/t/newbie-question-when-to-return-a-pointer/792/9
https://enterprisecraftsmanship.com/posts/entity-vs-value-object-the-ultimate-list-of-differences/
https://stackoverflow.com/questions/75446/value-vs-entity-objects-domain-driven-design
https://www.ardanlabs.com/blog/2014/12/using-pointers-in-go.html

Passing Entity types by values creates some practical problems (lost changes, accidental copies, some sync types not supporting copying, etc), but also some conceptual problems (Entities are just not something that must be ever copied, e.g. an object representing a database, if you draw a system diagram you want it to be a single box on the diagram; as opposed to ints/strings/Time that are Value types).

Reply:
Thank you very much for the clarification!